odoo/addons/base/models/assetsbundle.py

```diff
@@ -377,47 +402,57 @@ def is_css_preprocessed(self):
             Returns the bundle's flat css.
         """
-        for atype in (SassStylesheetAsset, ScssStylesheetAsset, LessStylesheetAsset):
-            assets = [asset for asset in self.stylesheets if isinstance(asset, atype)]
-            if assets:
-                source = '\n'.join([asset.get_source() for asset in assets])
-                compiled = self.compile_css(assets[0].compile, source)
-
-                if not self.css_errors and old_attachments:
-                    old_attachments.unlink()
-
-                fragments = self.rx_css_split.split(compiled)
-                at_rules = fragments.pop(0)
-                if at_rules:
-                    # Sass and less moves @at-rules to the top in order to stay css 2.1 compatible
-                    self.stylesheets.insert(0, StylesheetAsset(self, inline=at_rules))
-                while fragments:
-                    asset_id = fragments.pop(0)
-                    asset = next(asset for asset in self.stylesheets if asset.id == asset_id)
-                    asset._content = fragments.pop(0)
-
-                    if debug:
-                        try:
-                            fname = os.path.basename(asset.url)
-                            url = asset.html_url
-                            with self.env.cr.savepoint():
-                                self.env['ir.attachment'].sudo().create(dict(
-                                    datas=base64.b64encode(asset.content.encode('utf8')),
-                                    mimetype='text/css',
-                                    type='binary',
-                                    name=url,
-                                    url=url,
-                                    datas_fname=fname,
-                                    res_model=False,
-                                    res_id=False,
-                                ))
-
-                            if self.env.context.get('commit_assetsbundle') is True:
-                                self.env.cr.commit()
-                        except psycopg2.Error:
-                            pass
+        if self.stylesheets:
+            compiled = ""
+            for atype in (SassStylesheetAsset, ScssStylesheetAsset, LessStylesheetAsset):
+                assets = [asset for asset in self.stylesheets if isinstance(asset, atype)]
+                if assets:
+                    source = '\n'.join([asset.get_source() for asset in assets])
+                    compiled = self.compile_css(assets[0].compile, source)
+
+            # We want to run rtlcss on normal css, so merge it in compiled
+            if self.user_direction == 'rtl':
+                stylesheet_assets = [asset for asset in self.stylesheets if not isinstance(asset, (SassStylesheetAsset, ScssStylesheetAsset, LessStylesheetAsset))]
+                compiled += '\n'.join([asset.get_source() for asset in stylesheet_assets])
+                compiled = self.run_rtlcss(compiled)
```

Comment:
The branch is not rebased. There has been changes in master that can impact this. I think this should be moved at the very end of `compile_css` function.

Reply:
@qsm-odoo Tested with debug=assets and it works fine, about css file yes we have css files available in odoo, account, point_of_sale, pos_restaurant are few examples.
btw, don't you think we should support conversion of css files for right to left even if we do not have css file in future